### DISCUSSION TITLE
Can skyfield compute the .at() function for a list of timestamps/datetimes?

### DISCUSSION THREAD

**Discussion by pablotejada** [skyfielders/python-skyfield#865] [Q&A]
I am simulating a MEO constellation for the course of N years. Now I only have one earth location to compute the Lines of Sight (LoS) but it takes too long to compute, even with multiprocessing in place. Probably, is because I run the following code for every timestamp. Is there a way to compute the `.at()` in bulk for a list of timestamps, something that leverages NumPy?

This is the basic structure of the current code:

```
satellite : EarthSatellite
ground_pos : wgs84.latlon(lat, lon)

difference = satellite-ground_pos
topocentric = difference.at(t) #where t is a single time date
```

Ideally, it would be something like `difference.at([t1,t2,...,tm])` that allowed to compute the difference at all these times in bulk without having to iterate every timestamp, which is very slow. If this was possible, N-satellite to M-gateways system simulations would be faster to run, which will be great for the New-Space era! 

Thank you in advance!

P.S: Skyfield has been a great finding and I love it! @brandon-rhodes

**Comment by brandon-rhodes** [comment]
> P.S: Skyfield has been a great finding and I love it!

Thank you for the kind words, and I'm glad that you have been enjoying Skyfield!

Have you tried out Skyfield time vectors, which place _n_ times in the same `Time` object? Here's some documentation about them:

https://rhodesmill.org/skyfield/time.html#date-arrays

The underlying `sgp4` library that Skyfield uses to compute satellite positions also supports array operations, if you wanted to generate raw satellite coordinates and then perform the coordinate math to make them usable for your application — which is more complicated to work out, but also likely faster:

https://pypi.org/project/sgp4/#array-acceleration

**Reply by pablotejada** [reply]
Thank you so much @brandon-rhodes the first [link](https://rhodesmill.org/skyfield/time.html#date-arrays) solved my issue perfectly! Literally running simulations ~50 times faster now.

Have a great week!
